### PR TITLE
Remove unnecessary fields from DKG Private Info

### PR DESCRIPTION
**Pull request by tarakby**
The stored DKG private data should be just the Random Beacon key, while the implementation stores the node ID and the node index (in DKG and threshold signature). Both extra information can be found elsewhere: 
 - node ID is stored in `Local`
 - DKG index (and all node indices) are stored in the protocol state.

### CODE REVIEW

**Review by zhangchiqing — APPROVED** [review]
Nice!

File: storage/badger/dkg_keys.go [hunk 1]
```diff
@@ -27,15 +27,15 @@ func NewDKGKeys(collector module.CacheMetrics, db *badger.DB) (*DKGKeys, error)
 
 	store := func(key interface{}, val interface{}) func(*transaction.Tx) error {
```
Comment (jordanschalm) [inline storage/badger/dkg_keys.go:28]: We could change the name of this storage interface to `BeaconKeys` to match the other name changes

File: storage/badger/dkg_keys.go [hunk 2]
```diff
@@ -52,25 +52,25 @@ func NewDKGKeys(collector module.CacheMetrics, db *badger.DB) (*DKGKeys, error)
...
-func (k *DKGKeys) InsertMyDKGPrivateInfo(epochCounter uint64, info *dkg.DKGParticipantPriv) error {
+func (k *DKGKeys) InsertMyBeaconPrivateKey(epochCounter uint64, info *encodable.RandomBeaconPrivKey) error {
 	return operation.RetryOnConflictTx(k.db, transaction.Update, k.storeTx(epochCounter, info))
 }
 
-func (k *DKGKeys) RetrieveMyDKGPrivateInfo(epochCounter uint64) (*dkg.DKGParticipantPriv, error) {
+func (k *DKGKeys) RetrieveMyBeaconPrivateKey(epochCounter uint64) (*encodable.RandomBeaconPrivKey, error) {
 	tx := k.db.NewTransaction(false)
 	defer tx.Discard()
 	return k.retrieveTx(epochCounter)(tx)
```
Comment (jordanschalm) [inline storage/badger/dkg_keys.go:76]: could rename this file to `beacon_keys.go`

File: storage/badger/dkg_keys_test.go [hunk 3]
```diff
@@ -36,21 +36,21 @@ func TestDKGKeysInsertAndRetrieve(t *testing.T) {
...
+		actual, err := store.RetrieveMyBeaconPrivateKey(epochCounter)
 		require.NoError(t, err)
 		assert.Equal(t, expected, actual)
 
 		// test storing same key
-		err = store.InsertMyDKGPrivateInfo(epochCounter, expected)
+		err = store.InsertMyBeaconPrivateKey(epochCounter, expected)
 		require.True(t, errors.Is(err, storage.ErrAlreadyExists))
 	})
 }
```
Comment (jordanschalm) [inline storage/badger/dkg_keys_test.go:56]: rename file

File: storage/badger/operation/dkg_keys.go [hunk 4]
```diff
@@ -3,13 +3,13 @@ package operation
...
-func InsertMyDKGPrivateInfo(epochCounter uint64, info *dkg.DKGParticipantPriv) func(*badger.Txn) error {
-	return insert(makePrefix(codeDKGPrivateInfo, epochCounter), info)
+func InsertMyBeaconPrivateKey(epochCounter uint64, info *encodable.RandomBeaconPrivKey) func(*badger.Txn) error {
+	return insert(makePrefix(codeBeaconPrivateKey, epochCounter), info)
 }
 
-func RetrieveMyDKGPrivateInfo(epochCounter uint64, info *dkg.DKGParticipantPriv) func(*badger.Txn) error {
-	return retrieve(makePrefix(codeDKGPrivateInfo, epochCounter), info)
+func RetrieveMyBeaconPrivateKey(epochCounter uint64, info *encodable.RandomBeaconPrivKey) func(*badger.Txn) error {
+	return retrieve(makePrefix(codeBeaconPrivateKey, epochCounter), info)
```
Comment (jordanschalm) [inline storage/badger/operation/dkg_keys.go:14]: rename file

File: storage/dkg_keys.go [hunk 5]
```diff
@@ -1,10 +1,10 @@
 package storage
 
 import (
-	"github.com/onflow/flow-go/model/dkg"
+	"github.com/onflow/flow-go/model/encodable"
 )
 
 type DKGKeys interface {
```
Comment (jordanschalm) [inline storage/dkg_keys.go:7]: rename interface and file

**Review by jordanschalm — APPROVED** [review]
🎸  Nice work

Just some suggestions to extend the rename to a few more files/components

**Comment by tarakby** [comment]
bors merge

**Pull request merged**